### DISCUSSION TITLE
Unable to calculate buckling using Ideastatica.plugin

### DISCUSSION THREAD

**Discussion by Idonotuse** [idea-statica/ideastatica-public#714] [Help Q&A]
Hi,

I'm working on automating report generation using the `Ideastatica.Plugin`. The report needs to include buckling results; however, even after incorporating `CalculateBuckling` in the code, the generated report does not display any buckling results. I would appreciate your assistance in resolving this issue.

```
string Install_dir = "C:\\Program Files\\IDEA StatiCa\\StatiCa 24.1";
string report = report_path;

ConnHiddenClientFactory calcFactory = new ConnHiddenClientFactory(Install_dir);

ConnectionHiddenCheckClient checkClient = calcFactory.Create();

checkClient.OpenProject(Idea_file_path);

ConProjectInfo conProjectInfo = checkClient.GetProjectInfo();

string Conn_ID = conProjectInfo.Connections.First().Identifier;

ConnReportSettings settings = new ConnReportSettings();
     
checkClient.CalculateBuckling(Conn_ID);

checkClient.GenerateReportPdf(Conn_ID, report, settings);


checkClient.CloseProject();
```

Moreover, the behavior of same code is different for different versions. 

![image](https://github.com/user-attachments/assets/aa0db83a-46f7-4957-a09c-fbbaf092662c)

![image](https://github.com/user-attachments/assets/76eed777-3f06-4d25-9b24-e44e3ace2ee8)

**Comment by nathanlukeIDEA** [comment]
Hi @Idonotuse, 

Apologies for the delayed response. I have tested and yes you are correct, the buckling appears to not be calculated when printing the report through API. We are looking into fixing it now. 

The issue you mention related to desktop app 24.1 is that we have now released a new Connection API and we encourage you to update to get the calculation working.  More information and examples can be found here: https://github.com/idea-statica/ideastatica-public/tree/main/src/api-sdks/connection-api/clients/csharp/examples/CodeSamples

Kind regards
Nathan

**Reply by Idonotuse** [reply]
Hi @nathanlukeIDEA,

Any update on the above issue? 

I've also checked with the new connection API (V24.1.2.1513) and tried to generate report (using this: https://github.com/idea-statica/ideastatica-public/blob/main/src/api-sdks/connection-api/clients/csharp/examples/CodeSamples/Samples/GeneratePdf.cs), but the generated report doesn't have the buckling results.

**Reply by nathanlukeIDEA** [reply]
Hi @Idonotuse, we are still working on including buckling results in the report when calculated through the API.
